packager/mpd/base/period_unittest.cc

```diff
@@ -316,10 +316,12 @@ TEST_F(PeriodTest, TrickPlayWithNoMatchingAdaptationSet) {
             testable_period_.GetOrCreateAdaptationSet(
                 ConvertToMediaInfo(kVideoMediaInfo),
                 content_protection_in_adaptation_set_));
-  // A nullptr is returned if it is not able to find matching AdaptationSet.
-  ASSERT_FALSE(testable_period_.GetOrCreateAdaptationSet(
+
+  ASSERT_TRUE(testable_period_.GetOrCreateAdaptationSet(
       ConvertToMediaInfo(kVp9TrickPlayMediaInfo),
       content_protection_in_adaptation_set_));
+
+  ASSERT_TRUE(!testable_period_.GetTrickPlayCache().empty());
 }
 
```

Comment:
Can you add another unit-test similar to TrickPlayWithMatchingAdaptationSet but with operation at line 272 and 276 reversed to test the new code?